cirq-core/cirq/sim/simulator.py

```diff
@@ -836,15 +836,17 @@ def sample_measurement_ops(
         """
 
         # Sanity checks.
-        seen_measurement_keys: Set[str] = set()
         for op in measurement_ops:
             gate = op.gate
             if not isinstance(gate, ops.MeasurementGate):
                 raise ValueError(f'{op.gate} was not a MeasurementGate')
-            key = protocols.measurement_key_name(gate)
-            if key in seen_measurement_keys:
-                raise ValueError(f'Duplicate MeasurementGate with key {key}')
-            seen_measurement_keys.add(key)
+        result = collections.Counter(
+            key for op in measurement_ops for key in protocols.measurement_key_names(op)
+        )
+        if result:
+            duplicates = [k for k, v in result.most_common() if v > 1]
+            if duplicates:
+                raise ValueError(f"Measurement key {','.join(duplicates)} repeated")
```

Comment:
[No change needed] This formatting is ambiguous (`a,b` is a valid measurement key), but comparing with the circuit should be enough for it not to be an issue.